nengo_viz/swi.py

```diff
@@ -402,14 +408,18 @@ def start(cls, port=80, asynch=True, addr='', browser=False):
             # shut down any remaining threads
             if asynch and server.requests is not None:
                 for _, sock in server.requests:
```

Comment:
Actually, looking at your implementation down below, if list copying is atomic (and I think it is), then all the lock stuff is unnecessary.